script/deploy/deploy.ts

```diff
@@ -291,8 +295,9 @@ const deployControllerChainContracts = async (
       deployParams
     );
 
-    deployParams.addresses[SuperBridgeContracts.Controller] =
-      controller.address;
+    deployParams.addresses[SuperBridgeContracts.Controller] = getDryRun()
+      ? AddressZero
+      : controller.address;
```

Comment:
with this pattern, the stored addresses in json file would get overwritten with zero addresses if someone runs with dryrun.

Reply:
yup, that happened indeed. Fixed this by checking in the function for dryrun